### PR TITLE
Change highlighter, fetch value only in source_ and add deprecation warnings

### PR DESCRIPTION
**Pull request by chiragjn**
**Major Change**: We will now use the [unified highlighter](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-highlighting.html#unified-highlighter) as it much faster for larger documents.

This will make ES search queries faster and to be able to support a large list of variants per value as we are only concerned with highlights when searching

Also added deprecation warnings and FIXME notes for older code

### CODE REVIEW

File: datastore/datastore.py [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import warnings
```
Comment (srajdev) [inline datastore/datastore.py]: W1618 [34mimport missing `from __future__ import absolute_import`[0m

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Comment by haptik-deployment** [comment]
UNIT TESTS FAILED... Please resolve issues

**Comment by haptik-deployment** [comment]
UNIT TESTS FAILED... Please resolve issues

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Pull request merged**